From be15c1793fc8c08fd7105b8a9758bb50aa7546b1 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Mon, 9 Aug 2021 21:53:46 +0800
Subject: [PATCH 12/12] LoongArch: make .text start at 0x10000

The previous value obviously is coming from MIPS; no reason is given as
to why this is absolutely needed.  Just change to get closer to RISC-V
and a few other arches.

2021-10-25  WANG Xuerui <git@xen0n.name>

ld/
        * emulparams/elf64loongarch-defs.sh: Change TEXT_START_ADDR to
        0x10000.
---
 ld/emulparams/elf64loongarch-defs.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ld/emulparams/elf64loongarch-defs.sh b/ld/emulparams/elf64loongarch-defs.sh
index c793f5d8388..744b018a6f6 100644
--- a/ld/emulparams/elf64loongarch-defs.sh
+++ b/ld/emulparams/elf64loongarch-defs.sh
@@ -30,7 +30,7 @@ esac
 # LoongArch nop is 'andi $r0,$r0,0'.
 NOP=0x00004003
 
-TEXT_START_ADDR=0x120000000
+TEXT_START_ADDR=0x10000
 MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
 COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
 
-- 
2.33.1

